src/css_parser.rs

```diff
@@ -995,6 +1001,110 @@ fn parse_color_rgb_components<'a>(components: &mut Iterator<Item = &'a str>)
     })
 }
 
+/// Parse a color of the form 'hsl([0.-360.]deg, [0-100]%, [0-100]%)', or 'hsla([0.-360.]deg, [0-100]%, [0-100]%, [0.-1.])' without the leading 'hsl[a](' or trailing ')'. Alpha defaults to 255.
+fn parse_color_hsl<'a>(input: &'a str, parse_alpha: bool)
+-> Result<ColorU, CssColorParseError<'a>>
+{
+    let mut components = input.split(',').map(|c| c.trim());
+    let rgb_color = parse_color_hsl_components(&mut components)?;
+    let a = if parse_alpha {
+        parse_alpha_component(&mut components)?
+    } else {
+        255
+    };
+    if let Some(arg) = components.next() {
+        return Err(CssColorParseError::ExtraArguments(arg));
+    }
+    Ok(ColorU { a, ..rgb_color })
+}
+
+/// Parse the color components passed as arguments to an hsl(...) CSS color.
+fn parse_color_hsl_components<'a>(components: &mut Iterator<Item = &'a str>)
+-> Result<ColorU, CssColorParseError<'a>>
+{
+    #[inline]
+    fn angle_from_str<'a>(components: &mut Iterator<Item = &'a str>, which: CssColorComponent)
+    -> Result<f32, CssColorParseError<'a>>
+    {
+        let c = components.next().ok_or(CssColorParseError::MissingColorComponent(which))?;
+        if c.is_empty() {
+            return Err(CssColorParseError::MissingColorComponent(which));
+        }
+        let dir = parse_direction(c)?;
+        match dir {
+            Direction::Angle(deg) => Ok(deg.get()),
+            Direction::FromTo(_, _) => return Err(CssColorParseError::UnsupportedDirection(c)),
+        }
+    }
+
+    #[inline]
+    fn percent_from_str<'a>(components: &mut Iterator<Item = &'a str>, which: CssColorComponent)
+    -> Result<f32, CssColorParseError<'a>>
+    {
+        let c = components.next().ok_or(CssColorParseError::MissingColorComponent(which))?;
+        if c.is_empty() {
+            return Err(CssColorParseError::MissingColorComponent(which));
+        }
+        match parse_percentage(c) {
+            Some(percentage) => Ok(percentage.get()),
+            None => Err(CssColorParseError::InvalidPercentage(c)),
+        }
+    }
+
+    /// Adapted from [https://en.wikipedia.org/wiki/HSL_and_HSV#Converting_to_RGB]
+    #[inline]
+    fn hsl_to_rgb<'a>(h: f32, s: f32, l: f32) -> (u8, u8, u8) {
+        let s = s / 100.;
```

Comment:
Please use `.0` for floating-point numbers, this way it's easier to see that they are actually floating point and not integers. I know `100.0` is the same as a `100.`, but it makes things more clear.